include/string_utils.h

```diff
@@ -276,7 +277,8 @@ std::vector<std::string> split(std::string_view seq, char delim);
 //   split("a\tb\nc\vd e\rf") returns {"a", "b", "c", "d", "e", "f"}
 //   split("  ") returns {}
 //   split(" ") returns {}
-std::vector<std::string> split(std::string_view seq);
+std::vector<std::string> split(std::string_view seq,
+                               std::string_view delims = " \f\n\r\t\v");
```

Comment:
This will mean we have two "split" routines with virtually identical interfaces (char* vs. string_view, which itself can accept a char*) but with different behavior:


![2023-09-12_09-41](https://github.com/dosbox-staging/dosbox-staging/assets/1557255/5011812a-dbe1-4bd9-a503-1de0bdddb8ff)

---

![2023-09-12_09-40](https://github.com/dosbox-staging/dosbox-staging/assets/1557255/b3d55dff-5f00-4bf9-98aa-771de3c09d49)

---

We should only have one API:
 - `split(haystack, delims)`
 - `split(haystack)` <- simply calls `split(haystack, " \f\n\r\t\v")`

Can you check the existing use-cases of the `split(..)`. and see if you can boil it down to one implementation? (I really don't want to have two split rountines with different output)

Reply:
Yes, just keep this new one and get rid of the old one that accepts a single char delimiter (and refactor accordingly, please).

Reply:
I like your new rename_with_empties name. Let's go with the rename, and yours becomes the better default split() 👍